bin/proving-service/src/proxy/status.rs

```diff
@@ -0,0 +1,128 @@
+use std::{sync::Arc, time::Duration};
+
+use pingora::{
+    apps::HttpServerApp,
+    http::ResponseHeader,
+    prelude::*,
+    protocols::{Stream, http::ServerSession},
+    server::ShutdownWatch,
+};
+use serde::Serialize;
+use tonic::{async_trait, transport::Channel};
+use tracing::error;
+
+use super::worker::WorkerHealthStatus;
+use crate::{
+    commands::worker::ProverType, error::ProvingServiceError,
+    generated::status::status_api_client::StatusApiClient, proxy::LoadBalancerState,
+};
+
+/// Create a gRPC [StatusApiClient] for the given worker address.
+///
+/// # Errors
+/// - [ProvingServiceError::InvalidURI] if the worker address is invalid.
+/// - [ProvingServiceError::ConnectionFailed] if the connection to the worker fails.
+pub async fn create_status_client(
+    address: String,
+    connection_timeout: Duration,
+    total_timeout: Duration,
+) -> Result<StatusApiClient<Channel>, ProvingServiceError> {
```

Comment:
As mention in one of the previous comments, this function could probably be a helper function in `worker.rs`.